src/main/java/com/faforever/client/logging/analysis/AnalysisResult.java

```diff
@@ -0,0 +1,11 @@
+package com.faforever.client.logging.analysis;
+
+import com.faforever.client.notification.Action;
+
+import java.util.Collection;
+
+public record AnalysisResult(Collection<String> analysisMessages, Collection<Action> actions) {
+  public boolean isOk() {
+    return null == analysisMessages || analysisMessages.isEmpty();
+  }
+}
```

Comment:
analysiisMessages and actions should be non null since they are collections

Reply:
Additionally how it is implemented the action is tied to a specific message but how you have it implemented that is not clear and they cannot be displayed in a manner that makes it clear to the use. One would probably want to tie the action and message more closely together.